### PR TITLE
feat(docs): add accessibility section for Table component

### PR DESCRIPTION
**Pull request by chida09**
Closes https://github.com/yamada-ui/yamada-ui/issues/1575

## Description

in English
スクリーンショット 2024-10-21 21 57 41

スクリーンショット 2024-10-23 22 28 41

スクリーンショット 2024-10-23 22 29 57

in Japanese
スクリーンショット 2024-10-21 21 58 12

スクリーンショット 2024-10-23 22 28 26

スクリーンショット 2024-10-23 22 30 12

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Comment by zonoryo17** [comment]
@chida09 
badge

Is the `aria-label` item not necessary?

**Review by zonoryo17 — COMMENTED** [review]
@chida09
badge

Is the aria-label item not necessary?

**Comment by chida09** [comment]
@zonoryo17 
Based on my observation of other components, I noticed that the aria-label attribute is not explicitly listed in the ARIA Roles and Attributes sections. I concluded that it's sufficient if aria-label is mentioned at the beginning of the accessibility guidelines for a component. This assessment is supported by examples from the Yamada UI documentation, such as:

https://yamada-ui.com/components/forms/toggle/accessibility
https://yamada-ui.com/components/forms/file-input/accessibility

In these examples, we can see that the aria-label is not included in the specific ARIA Roles and Attributes list, but it is mentioned in the general accessibility guidelines for the component.

**Comment by zonoryo17** [comment]
@chida09 

I see. Thank you for explaining in detail.
You are certainly correct.
Thank you for your response.

**Comment by zonoryo17** [comment]
@chida09 

I wonder if perhaps we should add a section on keyboard operation?
In the descriptions of other components, some include keyboard operation items while others do not.
(This might not be necessary at the moment, as the keyboard functionality in the Table component is not fully implemented.)

**Comment by chida09** [comment]
@zonoryo17 
As this PR has been merged, I will address it again.
https://github.com/yamada-ui/yamada-ui/pull/3112

And this one as well.
https://github.com/yamada-ui/yamada-ui/pull/3145

**Comment by hirotomoyamada** [comment]
@zonoryo17 

Please refer to #3145, #3112 to update the new keyboard operation and ARIA role attributes.

**Comment by chida09** [comment]
@hirotomoyamada 
I have updated the documentation regarding keyboard operations.
Commit URL: https://github.com/yamada-ui/yamada-ui/pull/3038/commits/826fff6bad9588587e6a9cf17b04f768d0f1db1a
Reference document: https://github.com/yamada-ui/yamada-ui/blob/main/docs/contents/components/disclosure/accordion/accessibility.ja.mdx#L12-L13

There's one point I'd like to confirm: Is it intentional that pressing the Enter key on a table's checkbox does not activate the check?

Current behavior of the Yamada UI Table component:
https://yamada-ui.com/components/data-display/table

In the React Aria Table component, the Enter key activates the checkbox:
https://react-spectrum.adobe.com/react-aria/Table.html#table

**Comment by zonoryo17** [comment]
@hirotomoyamada 

> Please refer to https://github.com/yamada-ui/yamada-ui/pull/3145, https://github.com/yamada-ui/yamada-ui/pull/3112 to update the new keyboard operation and ARIA role attributes.

Thank you for sending me the link to the relevant PR.
Thank you again for your response.

**Comment by hirotomoyamada** [comment]
@chida09 

The default for `checkbox` does not support selecting with the `enter` key. So we have not adapted that.

**Comment by chida09** [comment]
I understand.
There are still some areas that need revision, so I will request another review once I have updated it.

File: docs/contents/components/data-display/table/accessibility.ja.mdx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## ARIAロールと属性
+
+| コンポーネント          | ロールと属性    | 使い方                                                                                                          |
+| ----------------------- | --------------- | --------------------------------------------------------------------------------------------------------------- |
+| `Table`                 | `role="table"`  | テーブルであることを示します。                                                                                  |
+|                         | `aria-disabled` | `disabledRowIds`が設定されている場合は、`true`を設定します。                                                    |
+|                         | `aria-rowcount` | テーブルの行数を示します。                                                                                      |
+|                         | `aria-colcount` | テーブルの列数を示します。                                                                                      |
+| `Thead` <Tag>内部</Tag> | `aria-sort`     | 表の項目が昇順の場合は`ascending`を設定し、降順の場合は`descending`を設定、指定がない場合は`none`を設定します。 |
```
Comment (hirotomoyamada) [inline docs/contents/components/data-display/table/accessibility.ja.mdx:39]: Change `Thead` to `Th`.

File: docs/contents/components/data-display/table/accessibility.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## ARIA Roles and Attributes
+
+| Component                   | Role and Attributes | Usage                                                                                                   |
+| --------------------------- | ------------------- | ------------------------------------------------------------------------------------------------------- |
+| `Table`                     | `role="table"`      | Indicates that the element is a table.                                                                  |
+|                             | `aria-disabled`     | Set to `true` if `disabledRowIds` is specified.                                                         |
+|                             | `aria-rowcount`     | Indicates the total number of rows in the table.                                                        |
+|                             | `aria-colcount`     | Indicates the total number of columns in the table.                                                     |
+| `Thead` <Tag>Internal</Tag> | `aria-sort`         | Set to `ascending` for ascending order, `descending` for descending order, and `none` if not specified. |
```
Comment (hirotomoyamada) [inline docs/contents/components/data-display/table/accessibility.mdx:39]: Change `Thead` to `Th`.

File: docs/contents/components/data-display/table/accessibility.mdx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+|                             | `aria-colindex`     | Indicates which column the element is in the table.                                                     |
+|                             | `aria-rowindex`     | Indicates which row the element is in the table.                                                        |
+| `Tbody` <Tag>Internal</Tag> | `aria-selected`     | Set to `true` if the row is selected, `false` if not selected.                                          |
```
Comment (hirotomoyamada) [inline docs/contents/components/data-display/table/accessibility.mdx:42]: Change `Tbody` to `Tr`.

File: docs/contents/components/data-display/table/accessibility.ja.mdx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+| `ArrowDown`     | 無効ではない次の項目をフォーカスします。                                                                         | -    |
+| `Home`          | 無効ではない最初の項目をフォーカスします。                                                                       | -    |
+| `End`           | 無効ではない最後の項目をフォーカスします。                                                                       | -    |
+
+## ARIAロールと属性
+
+| コンポーネント          | ロールと属性    | 使い方                                                                                                          |
+| ----------------------- | --------------- | --------------------------------------------------------------------------------------------------------------- |
+| `Table`                 | `role="table"`  | テーブルであることを示します。                                                                                  |
+|                         | `aria-disabled` | `disabledRowIds`が設定されている場合は、`true`を設定します。                                                    |
```
Comment (hirotomoyamada) [inline docs/contents/components/data-display/table/accessibility.ja.mdx:36]: Isn't `Tr` wrong, not `Table`?

Change "`disabledRowIds`が設定されている場合は、`true`を設定します。" to "行の`id`が`disabledRowIds`に含まれている場合は、`"true"`を設定します。".

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 1 above)
Comment (hirotomoyamada) [inline docs/contents/components/data-display/table/accessibility.ja.mdx:39]: Change "表の項目が昇順の場合は`ascending`を設定し、降順の場合は`descending`を設定、指定がない場合は`none`を設定します。" to "列が昇順の場合は`"ascending"`を設定し、降順の場合は`"descending"`を設定、指定がない場合は`"none"`を設定します。"

File: docs/contents/components/data-display/table/accessibility.ja.mdx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+|                         | `aria-rowcount` | テーブルの行数を示します。                                                                                      |
+|                         | `aria-colcount` | テーブルの列数を示します。                                                                                      |
+| `Thead` <Tag>内部</Tag> | `aria-sort`     | 表の項目が昇順の場合は`ascending`を設定し、降順の場合は`descending`を設定、指定がない場合は`none`を設定します。 |
+|                         | `aria-colindex` | 列がテーブルの何列目であるかを示します。                                                                        |
+|                         | `aria-rowindex` | 行がテーブルの何行目であるかを示します。                                                                        |
+| `Tbody` <Tag>内部</Tag> | `aria-selected` | 行が選択済みの場合は、`true`を設定し、未選択の場合は`false`を設定します。                                       |
```
Comment (hirotomoyamada) [inline docs/contents/components/data-display/table/accessibility.ja.mdx:42]: Change "行が選択済みの場合は、`true`を設定し、未選択の場合は`false`を設定します。" to "行が選択済みの場合は、`"true"`を設定し、未選択の場合は`"false"`を設定します。"

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 1 above)
Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.ja.mdx]: Fixed.
https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-2992dbf44702325b1d09cc77277848ac505ae7788a3fc76dbfda934dab7a5ff9R38-R40

https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-7ad307d16404c4e1c0bfc2852ffc14f4d55552661d2b9e66b357cee2c6012757R38-R40

File: docs/contents/components/data-display/table/accessibility.mdx (hunk 2 above)
Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.mdx]: Fixed.
https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-7ad307d16404c4e1c0bfc2852ffc14f4d55552661d2b9e66b357cee2c6012757R38-R40

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.ja.mdx]: Fixed.
https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-2992dbf44702325b1d09cc77277848ac505ae7788a3fc76dbfda934dab7a5ff9R38-R39

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 4 above)
Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.ja.mdx]: Fixed.
https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-7ad307d16404c4e1c0bfc2852ffc14f4d55552661d2b9e66b357cee2c6012757R38

https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-2992dbf44702325b1d09cc77277848ac505ae7788a3fc76dbfda934dab7a5ff9R38

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 1 above)
Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.ja.mdx]: Fixed.
https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-2992dbf44702325b1d09cc77277848ac505ae7788a3fc76dbfda934dab7a5ff9R40

https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-7ad307d16404c4e1c0bfc2852ffc14f4d55552661d2b9e66b357cee2c6012757R40

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 5 above)
Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.ja.mdx]: Fixed.
https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-2992dbf44702325b1d09cc77277848ac505ae7788a3fc76dbfda934dab7a5ff9R39

https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-7ad307d16404c4e1c0bfc2852ffc14f4d55552661d2b9e66b357cee2c6012757R39

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 1 above)
Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.ja.mdx]: Fixed.
https://github.com/yamada-ui/yamada-ui/pull/3038/files#diff-2992dbf44702325b1d09cc77277848ac505ae7788a3fc76dbfda934dab7a5ff9R40-R42

File: docs/contents/components/data-display/table/accessibility.ja.mdx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+`Table`は、アクセシビリティに関して[WAI-ARIA - Table Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/table/)に従います。
+
+`Table`に`aria-label`を設定すると、スクリーンリーダーによって読み上げられます。
+
+```tsx live=false
+<Table columns={columns} data={data} aria-label="DRAGON BALL characters" />
+```
+
+## キーボード操作
```
Comment (hirotomoyamada) [inline docs/contents/components/data-display/table/accessibility.ja.mdx:18]: The keyboard operation is fundamentally wrong.Please review everything.

File: docs/contents/components/data-display/table/accessibility.ja.mdx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+## ARIAロールと属性
+
+| コンポーネント       | ロールと属性    | 使い方                                                                                                          |
+| -------------------- | --------------- | --------------------------------------------------------------------------------------------------------------- |
+| `Table`              | `role="table"`  | テーブルであることを示します。                                                                                  |
+|                      | `aria-rowcount` | テーブルの行数を示します。                                                                                      |
+|                      | `aria-colcount` | テーブルの列数を示します。                                                                                      |
+| `Tr` <Tag>内部</Tag> | `aria-disabled` | 行の`id`が`disabledRowIds`に含まれている場合は、`"true"`を設定します。                                          |
+|                      | `aria-selected` | 行が選択済みの場合は、`"true"`を設定し、未選択の場合は`"false"`を設定します。                                   |
```
Comment (hirotomoyamada) [inline docs/contents/components/data-display/table/accessibility.ja.mdx:39]: Sorry, re-change "行が選択済みの場合は、`"true"`を設定し、未選択の場合は`"false"`を設定します。" to "行が選択されている場合は、`"true"`を設定し、未選択の場合は`"false"`を設定します。".

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 6 above)
Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.ja.mdx:18]: Should this be marked as 'Not applicable'?

>Keyboard Interaction
Not applicable.

https://www.w3.org/WAI/ARIA/apg/patterns/table/#keyboardinteraction

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 7 above)
Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.ja.mdx]: Fixed.
https://github.com/yamada-ui/yamada-ui/pull/3038/commits/02ab0d0cbd800ffd4425879c5f42d17bcb3ab792

File: docs/contents/components/data-display/table/accessibility.ja.mdx (hunk 6 above)
Comment (hirotomoyamada) [reply docs/contents/components/data-display/table/accessibility.ja.mdx:18]: Please refer here.

However, `Page Down`, `Page Up`, `Ctrl + Home` and `Ctrl + End` are optional and should be ignored.

https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/

Comment (chida09) [reply docs/contents/components/data-display/table/accessibility.ja.mdx:18]: Fixed.
https://github.com/yamada-ui/yamada-ui/pull/3038/commits/ebde4228a99c32eed75da83773e89593ffa5e707

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**